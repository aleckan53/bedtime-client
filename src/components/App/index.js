import React, { Component } from 'react'
import { Switch } from 'react-router-dom'
import Api from './api'
import Route from './components/Route'
import Nav from './components/Nav'
import {
  Auth, Editor, Home, LandingPage, Story, NotFound
} from '../'


export default class App extends Component {

  constructor(props) {
    super(props)
    this.state = {
      stories: [],
      error: null,
    }
  }

  componentDidMount() {
    Api.getStoriesList()
      .then(stories => this.setState({stories}))
      .catch(error => this.setState({error}))
  }

  render() {
    return (
      <div className='App'>
        <Route path='/app' component={Nav}/>
        <main>
          <Switch>
            <Route path='/app/story/:id' component={Story} stories={this.state.stories}/>
            <Route path='/app/auth' component={Auth}/>
            <Route path='/app/editor' component={Editor}/>
            <Route path='/app' component={Home} stories={this.state.stories}/>
            <Route exact path='/' component={LandingPage}/>
            <Route component={NotFound}/>
          </Switch>
        </main>
      </div>
    )
  }

}