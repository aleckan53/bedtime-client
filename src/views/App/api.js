import config from 'config'
import axios from 'axios'

export default {
  getStoriesList() {
    return axios.get(`${config.API_ENDPOINT}/stories`)
      .then(res => res.data)
      .catch(console.error)
  }
}