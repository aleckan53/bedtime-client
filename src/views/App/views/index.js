import Home from './Home'
import Editor from './Editor'
import Story from './Story'
import Authorization from './Authorization'

export {
  Home, Editor, Story, Authorization
}