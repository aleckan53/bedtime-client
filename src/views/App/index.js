import React, { useEffect, useState } from 'react'
import Api from './api'
import { Home, Editor, Story, Authorization } from './views'
import { Switch } from 'react-router-dom'
import { Route, Nav } from './components'


const App = props => {
  const [state, setState] = useState([])

  useEffect(() => {
    Api.getStoriesList()
      .then(setState)
  }, [])
  
  return (
    <div className='App'>
      <Nav/>
      <main>
        <Switch>
          <Route path='/app/story/:id' component={Story} stories={state}/>
          <Route path='/app/auth' component={Authorization}/>
          <Route path='/app/editor' component={Editor}/>
          <Route path='/app' component={Home} stories={state}/>
        </Switch>
      </main>
    </div>
  )
}

export default App
