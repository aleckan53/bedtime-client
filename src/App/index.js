import React, { useEffect, useState } from 'react'
import Api from './api'
import { Home, Editor } from 'views'
import { Switch } from 'react-router-dom'
import { Route, NavBar } from 'components/index'
import Story from 'views/Home/components/Story'


const App = props => {
  const [state, setState] = useState([])

  useEffect(() => {
    Api.getStoriesList()
      .then(setState)
  }, [])

  return (
    <div className='App'>
      <NavBar/>
      <main>
        <Switch>
          <Route path='/editor' component={Editor}/>
          <Route path='/home' component={Home} stories={state}/>
          <Route path='/story/:id' component={Story}/>
        </Switch>
      </main>
    </div>
  )
}

export default App
