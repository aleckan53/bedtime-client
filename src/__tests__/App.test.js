import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import App from '../components/App'
import { BrowserRouter } from 'react-router-dom'

describe(`App Component`, () => {

  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
      , div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders all routes', () => {
    const wrapper = shallow(<App />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})