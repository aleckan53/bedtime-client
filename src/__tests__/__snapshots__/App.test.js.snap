// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders all routes 1`] = `
<div
  className="App"
>
  <Route
    component={[Function]}
    path="/app"
  />
  <main>
    <Switch>
      <Route
        component={[Function]}
        path="/app/story/:id"
        stories={Array []}
      />
      <Route
        component={[Function]}
        path="/app/auth"
      />
      <Route
        component={[Function]}
        path="/app/editor"
      />
      <Route
        component={[Function]}
        path="/app"
        stories={Array []}
      />
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </main>
</div>
`;
